fullstory-segment-middleware/src/main/java/com/fullstorydev/fullstory_segment_middleware/FullStorySegmentMiddleware.java

```diff
@@ -0,0 +1,158 @@
+        this(context, segmentTag, new ArrayList<>());
+        Log.i(TAG, "no events allowlisted, unless enabled: allowlistAllTrackEvents, Segment track events will not be forward to FullStory!");
+    }
+
+    @Override
+    public void intercept(Chain chain) {
+        // Get the original payload from chain
+        BasePayload payload = chain.payload();
+
+        // create a place holder payload to be created
+        BasePayload newPayload = null;
+
+        // available payloads:
+        // https://github.com/segmentio/analytics-android/tree/master/analytics/src/main/java/com/segment/analytics/integrations
+        switch (payload.type()) {
+            case group:
+                GroupPayload groupPayload = (GroupPayload) payload;
+
+                // User will always be tied to the newest groupID that you set, group traits will not be included by default!
+                // this is because if the user changes it's group which no longer has certain traits, we will not detect when to clear them in FS and user will be tied to old group data
+                HashMap<String, Object> userVars = new HashMap<>();
+                userVars.put("groupID", groupPayload.groupId());
+                // optionally enable group traits to be passed into user vars
+                if (enableGroupTraitsAsUserVars && !isNullOrEmpty(groupPayload.traits())) {
+                    userVars.putAll(groupPayload.traits());
+                }
+                FS.setUserVars(userVars);
+                break;
+
+            case identify:
+                IdentifyPayload identifyPayload = (IdentifyPayload) payload;
+                FS.identify(identifyPayload.userId(), identifyPayload.traits());
+                break;
+
+            case screen:
+                ScreenPayload screenPayload = (ScreenPayload) payload;
+                if (this.enableSendScreenAsEvents) {
+                    FS.event("Segment Screen: " + screenPayload.name(), screenPayload.properties());
+                }
+                break;
+
+            case track:
+                TrackPayload trackPayload = (TrackPayload) payload;
+                if (this.allowlistAllTrackEvents || this.allowlistedEvents.indexOf(trackPayload.event()) != -1) {
+                    FS.event(trackPayload.event(), trackPayload.properties());
+                }
+                break;
+
+            default:
+                FS.log(FS.LogLevel.INFO, "FullStoryMiddleware: No matching API found, not calling any FS APIs...");
+                break;
+        }
+
+        Map<String, Object> context = new LinkedHashMap<>(payload.context());
+        if (this.enableFSSessionURLInEvents) {
+            newPayload = getNewPayloadWithFSURL(payload, context);
+        }
+
+        if (newPayload == null) newPayload = payload.toBuilder().context(context).build();
+        chain.proceed(newPayload);
+    }
+
+    private BasePayload getNewPayloadWithFSURL(BasePayload payload, Map<String, Object> context) {
```

Comment:
Similar to my request for the iOS middleware, can you include a comment about how the session replay URL is being used. I see that it decorates the context before sending along to the next middleware in the chain. How does this manifest itself in Segment?